xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -184,8 +184,14 @@ final class WrrSubchannel extends ForwardingSubchannel {
 
     @VisibleForTesting
     void onLoadReport(MetricReport report) {
-      double newWeight = report.getCpuUtilization() == 0 ? 0 :
-              report.getQps() / report.getCpuUtilization();
+      double newWeight = 0;
+      if (report.getCpuUtilization() > 0 && report.getQps() > 0) {
+        double penalty = 0;
+        if (report.getEps() > 0 && config.errorUtilizationPenalty > 0) {
+          penalty = report.getEps() / report.getQps() * config.errorUtilizationPenalty;
+        }
+        newWeight = report.getQps() / (report.getCpuUtilization() + penalty);
```

Comment:
if the cpu utilization is 0 and penalty is not, then the `newWeight` becomes 0, is that what we want?

Reply:
From my understanding, yes. Since originally with the old formula `weight = qps / cpu_utilization`, when `cpu_utilization` is 0, weight is 0. Would the case of `cpu_utilization` is 0 and penalty is not 0 ever happen?

Perhaps @yousukseung can clarify and confirm.

Reply:
It looks to me an edge case. The difference is that in the rare case (cpu_utilization is 0 and penalty is not 0), the channel gets an average weight or (qps/penalty) weight. Document the result if you decide to ignore that rare case and treat it as cpu utilization = 0?

Reply:
We should use the average weight when `cpu_utilization` is zero as indicated in the gRFC. Also confirmed C++ does that.

gRFC:
```
function UpdateWeight(qps, eps, cpu_utilization) {
  if (cpu_utilization > 0 && qps > 0)
    cpu_utilization += eps / qps * config.error_utilization_penalty;
  var new_weight =
     cpu_utilization == 0 ? 0 : qps / cpu_utilization;
  if (new_weight == 0) return;
  if (non_empty_since_ == infy) non_empty_since_ = now();
  last_updated_ = now();
  weight_ = new_weight;
}
```


C++:
```
float weight = 0;
if (qps > 0 && cpu_utilization > 0) {
  double penalty = 0.0;
  if (eps > 0 && error_utilization_penalty > 0) {
    penalty = eps / qps * error_utilization_penalty;
  }
  weight = qps / (cpu_utilization + penalty);
}
```

Reply:
Thanks for the clarification. The current Java code follows the same behaviour as C++ and the [gRFC A58](https://github.com/grpc/proposal/blob/master/A58-client-side-weighted-round-robin-lb-policy.md).

```
double newWeight = 0;
if (report.getCpuUtilization() > 0 && report.getQps() > 0) {
  double penalty = 0;
  if (report.getEps() > 0 && config.errorUtilizationPenalty > 0) {
    penalty = report.getEps() / report.getQps() * config.errorUtilizationPenalty;
  }
  newWeight = report.getQps() / (report.getCpuUtilization() + penalty);
}
if (newWeight == 0) {
  return;
}
if (nonEmptySince == infTime) {
  nonEmptySince = ticker.nanoTime();
}
lastUpdated = ticker.nanoTime();
weight = newWeight;
```